jax/_src/pjit.py

```diff
@@ -887,9 +888,120 @@ def _process_in_axis_resources(in_shardings_thunk, in_layouts_thunk, in_avals,
       for i, aval in zip(in_shardings_flat, in_avals))
   return canonicalized_shardings, tuple(in_layouts_flat)
 
+callsites: set[str] = set()
 
-@lu.cache
-def _create_pjit_jaxpr(fun, in_type, debug_info, out_paths):
+def explain_tracing_cache_miss(
+    f: Callable, unseen_f: bool, cache: dict, key: tuple, result: tuple):
```

Comment:
Is the `result` parameter being used? I couldn't find it's usage.

Reply:
No. I used it in an earlier iteration though, to print the jaxpr id being produced by the tracing cache. Might keep it in there in case we want it in the future... wdyt?

Reply:
Hmm, okay. Remind me, why would we need to print the jaxpr id? To connect the tracing cache with lowering cache?

Reply:
Yeah exactly. I found it useful when playing with things, just to connect the two error messages, since otherwise a compilation cache miss that happens right after a tracing cache miss could appear to be related when it's actually for a totally different computation. But since this PR doesn't have the compilation cache stuff in it, it's not relevant. It's not clear to me if it will be useful in the end...